pkg/xds/topology/router.go

```diff
@@ -60,26 +60,38 @@ func BuildRouteMap(dataplane *mesh_core.DataplaneResource, routes []*mesh_core.T
 	for _, oface := range dataplane.Spec.Networking.GetOutbound() {
 		serviceName := oface.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
 		policy, exists := policyMap[serviceName]
+
 		outbound := dataplane.Spec.Networking.ToOutboundInterface(oface)
 		if exists {
-			routeMap[outbound] = policy.(*mesh_core.TrafficRouteResource)
-		} else {
-			routeMap[outbound] = &mesh_core.TrafficRouteResource{
-				Meta: &pseudoMeta{
-					Name: "(implicit default route)",
-				},
-				Spec: mesh_proto.TrafficRoute{
-					Sources: []*mesh_proto.Selector{{
-						Match: mesh_proto.MatchAnyService(),
-					}},
-					Destinations: []*mesh_proto.Selector{{
-						Match: mesh_proto.MatchService(serviceName),
-					}},
-					Conf: []*mesh_proto.TrafficRoute_WeightedDestination{{
-						Weight:      100,
-						Destination: mesh_proto.MatchTags(oface.GetTagsIncludingLegacy()),
-					}},
-				},
+			route := policy.(*mesh_core.TrafficRouteResource)
+			if route.Spec.GetConf().HasWildcard() {
+				split := []*mesh_proto.TrafficRoute_Split{}
+				for _, destination := range route.Spec.GetConf().GetSplit() {
+					newDestination := make(map[string]string, len(destination.Destination))
+					for k, v := range destination.Destination {
+						if k == mesh_proto.ServiceTag {
+							v = serviceName
+						}
+						newDestination[k] = v
+					}
+					split = append(split, &mesh_proto.TrafficRoute_Split{
+						Weight:      destination.Weight,
+						Destination: newDestination,
+					})
+				}
+
+				routeMap[outbound] = &mesh_core.TrafficRouteResource{
+					Meta: route.GetMeta(),
+					Spec: mesh_proto.TrafficRoute{
+						Sources:      route.Spec.GetSources(),
+						Destinations: route.Spec.GetDestinations(),
+						Conf: &mesh_proto.TrafficRoute_Conf{
+							Split: split,
+						},
+					},
+				}
```

Comment:
I don't think that's the algorithm that we agreed on the call. The wildcard can be placed on many tags.

The idea was
1) Take traffic route
2) Remove all tag pairs with wildcard
3) Add all tags from outbound to traffic route if they do not already exist in the list

Examples
1) 
Traffic Route
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: '*'
```
+ Outbound
```
outbound:
- port: 1234
  tags:
    kuma.io/service: backend
```
=
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: 'backend'
```

2)
Traffic Route
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: '*'
      kuma.io/zone: aws
```
+ Outbound
```
outbound:
- port: 1234
  tags:
    kuma.io/service: backend
```
=
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: 'backend'
      kuma.io/zone: aws
```

3)
Traffic Route
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: '*'
      kuma.io/zone: '*'
```
+ Outbound
```
outbound:
- port: 1234
  tags:
    kuma.io/service: backend
```
=
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: 'backend'
```

4)
Traffic Route
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: '*'
      kuma.io/zone: '*'
```
+ Outbound
```
outbound:
- port: 1234
  tags:
    kuma.io/service: backend
    kuma.io/zone: aws
```
=
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: 'backend'
      kuma.io/zone: aws
```

5)
Traffic Route
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: '*'
```
+ Outbound
```
outbound:
- port: 1234
  tags:
    kuma.io/service: backend
    version: 1
```
=
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: 'backend'
      version: 1
```

6)
Traffic Route
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: '*'
      version: 1
```
+ Outbound
```
outbound:
- port: 1234
  tags:
    kuma.io/service: backend
    version: 2
```
=
```
conf:
  split:
  - weight: 100
    tags:
      kuma.io/service: 'backend'
      version: 1
```
(Traffic Route should take precedence)